### PR TITLE
feat(xy): expose projection area

### PR DESCRIPTION
**Pull request by markov00**
## Summary

Attaching a listener to the `<Settings onProjectionAreaChange={({projection,parent}) => void}` prop you can listen to parent and projection area changes.
The attached listener will receive an object with the dimensions (top, left, width, height) of both the projection area and the parent area.

Screenshot 2022-01-24 at 16 38 42

## Details

The exposed API is declared as:

```ts
type ProjectionAreaChangeListener = (areas: { projection: Dimensions; parent: Dimensions }) => void;
```

replace https://github.com/elastic/elastic-charts/pull/1539

### Checklist

- [x] The proper **chart type** label has been added (e.g. `:xy`, `:partition`)
- [x] The proper **feature** labels have been added (e.g. `:interactions`, `:axis`)
- [ ] All related issues have been linked (i.e. `closes #123`, `fixes #123`)
- [x] New public API exports have been added to `packages/charts/src/index.ts`
- [x] The proper documentation and/or storybook story has been added or updated

### CODE REVIEW

File: integration/server/webpack.config.js [hunk 1]
```diff
@@ -16,7 +16,7 @@ const webpack = require('webpack');
 
 module.exports = {
   entry: '../tmp/index.tsx',
-  mode: 'production',
+  mode: 'development',
```
Comment (walterra) [inline integration/server/webpack.config.js:19]: Not sure, is this change intentional here?

**Review by walterra — COMMENTED** [review]
Great update, looking forward to this. Overall LGTM, just one small question where I'm not sure if the change is intentional or accidental 😅

File: integration/server/webpack.config.js (hunk 1 above)
Comment (markov00) [reply integration/server/webpack.config.js]: thanks for the catch, fixed here https://github.com/elastic/elastic-charts/pull/1567/commits/9e8b85ec900a5699fe0b1190d3d5f9df889322b0

**Review by walterra — APPROVED** [review]
